src/SchemaParser.f90

```diff
@@ -0,0 +1,1551 @@
+        IF(.NOT.isSBKEntry(entry)) RETURN
+        CALL toUPPER(entry)
+        IF(entry=='TRUE' .OR. entry=='T') THEN
+          entrySBK(ientry)=.TRUE.
+        ELSE
+          entrySBK(ientry)=.FALSE.
+        ENDIF
+      ENDDO
+
+      !Add to parameter list
+      CALL paramList%add(TRIM(pListPath),entrySBK)
+
+    ENDSUBROUTINE parseSBKa1_SchEnt
+!
+!-------------------------------------------------------------------------------
+!> @brief Parsing routine for the given entry STRa1
+!> @param this         the entry to perform parsing
+!> @param entryStr     the entry string to be parsed
+!> @param paramList    the target parameter list in which to put the parsed data
+!> @param pListPathCrd the card parameter list path
+!>
+    SUBROUTINE parseSTRa1_SchEnt(this,entryStr,paramList,pListPathCrd)
+      CLASS(SchemaEntrySTRa1Type),INTENT(IN) :: this
+      CLASS(StringType),INTENT(IN) :: entryStr
+      CLASS(ParamType),INTENT(INOUT) :: paramList
+      TYPE(StringType),INTENT(IN) :: pListPathCrd
+
+      TYPE(StringType) :: pListPath
+      INTEGER(SIK) :: ierr,ientry
+      TYPE(StringType),ALLOCATABLE :: entry(:)
+
+      !Add to path
+      pListPath=pListPathCrd
+      CALL this%addPLPath(pListPath)
+
+      !Parse string into STRa1
+      ALLOCATE(entry(nFields(entryStr)))
+      DO ientry=1,nFields(entryStr)
+        CALL getField(ientry,entryStr,entry(ientry),ierr)
+      ENDDO
+
+      !Add to parameter list
+      CALL paramList%add(TRIM(pListPath),entry)
+
+    ENDSUBROUTINE parseSTRa1_SchEnt
+!
+!-------------------------------------------------------------------------------
+!> @brief Finds the index which corresponds to the provided element name
+!> @param elements  the element array to search for the given name
+!> @param name      the element name we are looking to match in elements
+!> @param index     the index number of the matching element in the array
+!>
+    FUNCTION findElementByName(elements,name) RESULT(index)
+      CLASS(SchemaElementType),INTENT(IN) :: elements(:)
+      TYPE(StringType),INTENT(IN) :: name
+      INTEGER(SIK) :: index
+
+      INTEGER(SIK) :: ielement
+
+      !Ensure the provided blockName matches a block name on this schema
+      index=UNDEFINED_ELEMENT
+      DO ielement=1,SIZE(elements)
+        IF(elements(ielement)%hasName(name)) THEN
+          index=ielement
+          EXIT
+        ENDIF
+      ENDDO
+    ENDFUNCTION findElementByName
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines if the input text file line is currently at End of the File
+!> @param file   the file currently being read
+!> @param isEOF  logical determining whether or not the file is at End of File
+!>
+    FUNCTION atEndOfFile(file) RESULT(isEOF)
+      TYPE(InputFileType),INTENT(IN) :: file
+      LOGICAL(SBK) :: isEOF
+
+      REQUIRE(file%isOpen())
+      isEOF=file%isEOF() .OR. file%getProbe()==DOT
+    ENDFUNCTION atEndOfFile
```

Comment:
Would be good to make this a method on the `InputFileType` instead

Reply:
I did consider this originally, but I thought that the DOT nomenclature to designate the end of a text file was something specific to MPACT and not to input files in general.  Though you could make the same argument for Schemas as well.

One solution is for Schema to have an End-of-File character that the user could optionally specify which defaults to DOT.

Reply:
Yeah, that's a good point.  I honestly think the dot thing is kind of pointless, but changing that is outside the scope of your changes, so let's leave this as is for now.